src/move_basic.cpp

```diff
@@ -614,70 +563,42 @@ bool MoveBasic::rotate(double yaw, const std::string& drivingFrame)
         double angleRemaining = requestedYaw - currentYaw;
         normalizeAngle(angleRemaining);
 
+        //ROS_INFO("%f", angleRemaining);
         double obstacle = collision_checker->obstacle_angle(angleRemaining > 0);
         double remaining = std::min(std::abs(angleRemaining), std::abs(obstacle));
-        double speed = std::max(minAngularVelocity,
-            std::min(maxAngularVelocity,
-              std::sqrt(2.0 * angularAcceleration *
-                (remaining - angularTolerance))));
-
-        double velocity = 0;
-
-        if (angleRemaining < 0.0) {
-            velocity = -speed;
-        }
-        else {
-            velocity = speed;
-        }
-
-        if (sign(prevAngleRemaining) != sign(angleRemaining)) {
-            oscillations++;
-        }
-        prevAngleRemaining = angleRemaining;
+        double velocity = std::max(-maxTurningVelocity,
+            std::min(rotGain*remaining, std::min(maxTurningVelocity,
+                    std::sqrt(2.0 * angularAcceleration *remaining))));
 
         if (actionServer->isPreemptRequested()) {
             ROS_INFO("MoveBasic: Stopping rotation due to preempt");
             done = true;
+            success = false;
             velocity = 0;
         }
 
-        //ROS_INFO("%f %f %f", rad2deg(angleRemaining), angleRemaining, velocity);
-
-        if (std::abs(angleRemaining) < angularTolerance || oscillations > 2) {
+        if (std::abs(angleRemaining) < angularTolerance) {
+            ROS_INFO("MoveBasic: Done rotation, error %f degrees", rad2deg(angleRemaining));
             velocity = 0;
+            success = true;
             done = true;
-            ROS_INFO("MoveBasic: Done rotation, error %f degrees", rad2deg(angleRemaining));
         }
+
+        bool reverse =  (angleRemaining < 0.0);
+        if (reverse) {
+            velocity = -velocity;
+        }
+
         sendCmd(velocity, 0);
     }
-    return done;
+    return success;
 }
 
 // Move forward specified distance
 
 bool MoveBasic::moveLinear(tf2::Transform& goalInDriving,
-                           const std::string& planningFrame,
```

Comment:
Is planning frame not needed anymore in move linear?

Reply:
I did thought about this overall, because there's lots of confusion about planning frame and driving frame, because they are actually the same for our application but didn't want to delete because it might make sense for other projects. Nevertheless, in moveLinear function it didn't make sense to have the planningFrame anymore because it was used for transformation that wasn't used. But I will definitely consider this in the future, because during linear portion of the goal, we could update the pose according to some frame other than /odom or /map and get a better localization(for example fiducial)